contrib/shibuqam/examples/shibuqamoauth.nit

```diff
@@ -0,0 +1,157 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Server that implements an authentication bound to a `shibuqam` valid server.
+#
+# # Use as a web service
+#
+# ## Request
+#
+# Two GET fields
+#
+# * `redir`: A full URI that will be used to redirect the user once the authentication is done.
+# * `state`: A temporary string used to check that the callback is legitimate.
+#
+# On request, the service authenticate the user then redirect to the callback with the user informations.
+#
+# ## Response
+#
+# After the authentication, the response is a redirection to `redir` with the following GET arguments:
```

Comment:
It agrees with me. No sensitive data is put in the URL. Only the authentication token.